server/src/main/java/org/opensearch/index/compositeindex/datacube/MetricStat.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.index.compositeindex.datacube;
+
+import org.opensearch.common.annotation.ExperimentalApi;
+
+/**
+ * Supported metric types for composite index
+ *
+ * @opensearch.experimental
+ */
+@ExperimentalApi
+public enum MetricStat {
+    COUNT("count"),
```

Comment:
We can have the metric stat type in the enum itself. Something like SUM will be of type DOUBLE

Reply:
This is customer facing and related to mappings, maybe we can have another enum which maps metric type to aggregation to have this data?